util: add numericSeparator to util.inspect (#41003)
nodejs/node: doc/api/util.md

    set to `'set'`, only getters with a corresponding setter are inspected.
    This might cause side effects depending on the getter function.
    **Default:** `false`.

>   * `numericSeparator` {boolean} If set to `true`, a underscore is used to

Reviewer: ```suggestion
  * `numericSeparator` {boolean} If set to `true`, an underscore is used to
```

>     separate thousands in all bigint and numbers. **Default:** `false`.

Reviewer: ```suggestion
    separate thousands in all bigints and numbers. **Default:** `false`.
```

>     separate thousands in all bigints and numbers. **Default:** `false`.

Reviewer 2: I'm OK with leaving this the way it is if we can't find a clear and concise way to more precisely explain it, but this isn't really thousands, right? It's every three digits in the left of the decimal point, so it separates the thousands place from the hundreds place, but also the millions place from the hundred-thousands place, and the billions place form the hundred-millions place, etc.

I'm having a hard time coming up with better wording at the moment though, and this is clear enough that most people will intuitively understand it, I think. So I'm fine with it landing as is and we can improve it later if it ends up causing any actual problems (which it probably won't) for readers.

That said, if you have a brilliant idea for how to make it more precise, please go for it!

Reviewer 3: What about "powers of thousand"?

Author: Suggestions are warmly welcome!

> every three digits to the left of the decimal point

It also separates the right part of the decimal point. E.g., `1_234_567.123_45`

Reviewer 2: In that case, I wonder if "every three digits" is sufficiently clear and simple.

```suggestion
    separate every three digits in all bigints and numbers. **Default:** `false`.
```

An example added somewhere nearby in the docs showing the output difference with this setting might be useful too.

Author: Done
